### ISSUE TITLE
Player's cannot hear each other over the radios

### ISSUE THREAD

**Issue by 5RBJS** [IDI-Systems/acre2#986]
**Mods:**
- **Arma 3:** `1.98`
- **CBA:** `3.15.1`
- **ACRE2:** `2.7.4`

**Description:**
Players cannot hear each other on the radios (152s and 117Fs) at random timings. It is as if there is mountain between us even if we are on flat terrain 200m away. I have altered the terrain interference settings and it hasn't fixed it. 

**Steps to reproduce:**
- Had a helicopter hover 300m above a player and neither the player nor the pilot could hear each other on a 152 or a 117F.
- Had two players stand looking at each other 20m away with a 152 and they couldn't hear each other (on the Altis Salt Lake)

**Expected behavior:**
Players should be able to hear other on the radios.

**Where did the issue occur?**
- Dedicated Server

**Log Files:**
- Game and Server Doesn't crash so there are no RPTs
- Extension Log File: https://pastebin.com/KceHhfWH

**Additional context:**
The issue appears to be random and is affected on both vanilla and modded maps (CUP and other). Sometimes it works fine, other times no one can hear each other at all.

Not sure if this was just a coincidence due to the random timing of it, but it seemed to be more common when more players were on the server.

**Comment by JustAKuba** [comment]
Yep, random, it looks like it is also related to radio type. Players didn't hear each other on 117F but could communicate on 152.

**Comment by 5RBJS** [comment]
During our tests, it occurred much more commonly with the 117F however I can confirm we have had the same issue on multiple occasions with the 152.

**Comment by Bradley-Silver** [comment]
Can also confirm we are also having the exact same issues as above and is driving me insane. Hope the fix will arrive soon.

**Comment by Montiemedia** [comment]
In our Group wie have the same issue.

**Comment by JustAKuba** [comment]
@jonpas @5RBJS @BradHex I was able to go around that issue by setting the terrain coefficient loss on lower value! Hope it will help.

**Comment by BroBeansCPG** [comment]
Have narrowed down an similar issue we're having in our group. 

Can you guys upload your server.rpt to see if I can match any similarities. I'm 90% sure what it is..

**Comment by Pironesis** [comment]
Have also encountered this over multiple missions and game sessions. Both on LOS Simple and Multipath, more commonly on multipath, we force ignoring antenna direction and alignment too. 

For some added clarity on what occurs players can be in direct line of sight, with either 152/117 radio and nothing can be heard. Not any kind of distortion from two people speaking over each other or from terrain interference, just silence. 

Also have had some instances of person X being able to hear person Y, but person Y cannot hear person X. With person Z able hear person X while being stood next to person Y.

**Comment by jonpas** [comment]
Any of you tried the dev-build 1030 that's available on our Slack yet?

**Comment by BroBeansCPG** [comment]
@Twelvecaliber can you upload your server .rpt mate?

**Comment by Pironesis** [comment]
[arma3server_x64_2020-07-04_18-01-26.txt](https://github.com/IDI-Systems/acre2/files/4948766/arma3server_x64_2020-07-04_18-01-26.txt)

@BroBeansCPG  
That's an RPT from a few weeks ago, when it was very prominent. The mission where it happens is in the RPT as  "co@ 34 Supporting Armour v01a" on line 4392. The issue was present throughout the entirety of the mission.  Also at the time we were on 2.7.3.1024 but have since had the same issues with 2.7.4.1027 i just can't remember an exact date with a concrete encounter in at the moment. 

As a side note, we do encounter this issue more often on the terrain Ruha, than on other terrains but not exclusively. Only thing of note that i can think of with Ruha is that it does have a higher heightmap resolution than most terrain, maybe that is somehow a factor.

**Comment by BroBeansCPG** [comment]
I seem to always have something similar to this line present in missions where comms ends up dropping out. 

I need to do some further investigation, but i believe it could be related 

`2020/07/04, 22:55:39 [ACRE] (sys_server) WARNING: Desync - Garbage collection prevented for radio 'acre_prc343_id_9' from player 'Tackleberry' as it still exists for them locally!`

**Comment by AaronMichaels** [comment]
Wanted to contribute;

- Using Lythium, Altis & Stratis
this issue still occurs regardless of map. Comms cut in and out randomly. Sometimes as Zeus If I teleport myself on top of a hill nearby some players can hear me but not all and it is unreliable.

It seems considerably worse on Lythium, maybe a foliage/object problem as well?


Also; @Twelvecaliber said >Also have had some instances of person X being able to hear person Y, but person Y cannot hear person X. With person Z able hear person X while being stood next to person Y.

I have also experienced the same with my group. As the Zeus operator, I'm normally able to hear my team leaders on our "black comms" but they rarely seem to be able to hear me.

**Comment by jonpas** [comment]
Can confirm Lythium has significantly worse quality.

Any of you tried the dev-build 1030 that's available on our Slack yet?

**Comment by BroBeansCPG** [comment]
Got a link handy? I can give it a test later on tonight

**Comment by Drofseh** [comment]
http://dev.idi-systems.com/dev/acre2_2.7.4.1030.zip

**Comment by Pironesis** [comment]
> Any of you tried the dev-build 1030 that's available on our Slack yet?

I have had two gamenights now on this version, so far no issues. Will keep checking, and try get a mission that was known to cause issues played this week.

**Comment by AaronMichaels** [comment]
@jonpas I intend on running a fun op on Wednesday (29th) with the dev-build Drofseh linked the other day. 

I'll come and report back with our experience! It likely won't be on Lythium to avoid confusing my guys with the storyline (they're simple lads)

**Comment by MadSoloSniper** [comment]
**MODS:**
Dedicated hard iron server (not a blade or shared space) in a Datacenter in KC
Our own TS Server (same box)
ArmA - 1.98.146373
TS - 3.5.3 (5/6/2020 - 04:07:50)
CBA - 3.15.1.200418
ACRE - 2.7.4.1027

[Arma 3.pdf](https://github.com/IDI-Systems/acre2/files/4995859/Arma.3.pdf)

**Description:**
Current map - G.O.S. N'ziwasogo v2 APEX

Doesn't matter radio (343. 152. 117) - total signal loss (no static, just full audio loss) after about 50M.
Clear LOS to person - we can actually almost hear each other in Direct while trying to comms on radio – recent test at the airfield southern end of the map – on the airfield (clear LOS) – we were able to relay between parties that were just less than ~50M away but any further and we experience total signal loss (no static, don’t even hear them key up).


**Steps to reproduce:**
Fire up the server and do it again (we did to make sure it wasn’t a single random issue)


**Expected behavior:**
Radio comms as typical, certainly within 100M of team when clear LOS is available
In the field expect regular ~500M range but with it not working WITH LOS and less than 100M – we’ve moved to straight TS for comms right now.


**Where did this occur?**
Dedicated hard iron server (not a blade or shared space) in a Datacenter in KC
Our own TS Server (same box)
Current map - G.O.S. N'ziwasogo v2 APEX


**Log files:**
N/A as there was no crash and ACRE thinks it’s running fine.

**Additional Context:**
Can be reproduced
All team members running latest version of the MODs, ARMA and TS

**_Second Issue:_**
Map - Lythium, Anizay, Altis, Malden, Kujari (the ones we have run on)
343 will at times work "ACROSS THE MAP" and will just as randomly provide significant 1x5 static when 100M away but in clear LOS to team member.

Happy to invite a member of the dev team to our Discord or to our server to experience it in real time.  Thanks for the dev and the help, hope we can sort this out and fix soon, I really hate using TS native as the comms function . . . . it just ruins my “immersion” . . . . LOL

**Comment by Nemesi-BDR** [comment]
We also experienced issues with 2.7.4, we reverted to 2.7.3 and it seems the issues are solved, at least so far.

**Comment by Montiemedia** [comment]
> We also experienced issues with 2.7.4, we reverted to 2.7.3 and it seems the issues are solved, at least so far.

Our Community did the same. Most of the problems are solved, but in 2.7.3 you have some connection issues too.

**Comment by MadSoloSniper** [comment]
While those are nice "work arounds" it doesn't address the issue of what happened and what the next steps (fix) will be? - Have never considered moving to another radio platform but this is more than just a "small issue" bug - this is a MAJOR break, would love to hear (know) what and when the solution will be, that will give me  the piece of mind just to hold on until it gets fixed vs. moving to another radio platform (not something that we would take lightly).

I know this is all "free time development", but shifting my team "back a version" would be a royal pain as they barely can keep MODs straight using the STEAM launcher (trust me), so the latest version (or no version like now) are my options.

I'll take any advice, thanks for the support and responses so far.

**Comment by jonpas** [comment]
Build 1030 (linked above) has some important fixes as mentioned and those issues were not yet encountered with it. If you want to test before release, that would be the one to test.

**Comment by MadSoloSniper** [comment]
Is there a link to the dev version on the STEAM workshop or am I having the team "manually" install this to test it?

PS - I'm really not talking "test" environment here, this is our production Mission Server - don't have an issue testing and reporting back, but my real question is what are the current dev plans for release?  Given that we are where we are (current release) that there is a plan for correcting what the latest version has done (broken) vs. what it actually usable.  Thanks in advance, and sorry to be a pain but we have some major operations coming up and I'm just trying to get this resolved in advance of those launches.

**Comment by jonpas** [comment]
No, it is not available on Steam. Plans are as always, released when ready and time is available to do so.

**Comment by MadSoloSniper** [comment]
Rough idea when the next update will be coming out (production release)??

Thanks,

Solo

**Comment by Pironesis** [comment]
Had another two sessions last week on 1030, only had one instance of issues where we could not talk with assets that were flying over us and such. However i am leaning more to the conclusion that instance was a mission side issue, as we have had missions on that terrain with no issues at all previously. 

Otherwise so far no issues on 1030 including on Ruha which for us was a often magnet for these comm failures.

**Comment by MadSoloSniper** [comment]
Awesome INTEL thank you - gives me some comfort taking the team to 1030 for now even while it's in DEV testing.  Good to know.

In the spirit of "team" I'll also report back as soon as we have some ground time to test.  We run a training / practice island on HEBONTES so that will get some serious testing and we're currently running operations on N'Zwag (APEX) so that will get some serious jungle testing.

Thanks again,
Solo

**Comment by Drofseh** [comment]
http://dev.idi-systems.com/dev/acre2_2.7.4.1032.zip

**Comment by Koch** [comment]
Just wanted to chime in and say that 1032 worked brilliantly (except for the one Zeus God Mode(?) issue I noted on Slack, which Jonpas is taking a look at).

No issues that we could discern after a session (with less than 10 people this time, will be trying with more next Sunday). If all works well we'll have more feedback with 50+ people in a week or two.

**Comment by MadSoloSniper** [comment]
Is there any update on when this is going into production?
Any signs of life are appreciated.
Solo

**Comment by TheMagnetar** [comment]
> Is there any update on when this is going into production?
> Any signs of life are appreciated.
> Solo

Typical answer: when it is ready.

**Comment by MadSoloSniper** [comment]
All good . . . 
Typical response: Moving to TFAR, might move back if / when things get better.

**Comment by Pironesis** [comment]
Ran into a rather catastrophic failure of ACRE on our gamenight on Saturday. To set the scene as it were, approx 24 players on the server, playing a mission on Ruha involving 6 aircraft and some ground elements, with most elements using 152 and 117 radios across multiple nets. Mission and server were both set to be using LOS Multipath and antenna direction ignored and server/clients using build 1032.

The progression of events were as follows:

At the start of the mission things were fine, extensive radio checks were done on the ground and all was well. After about 30-45 mins a single individual in an aircraft out of nowhere lost the ability to receive transmissions, other people could hear his outbound traffic fine, but he was unable to receive any responses to his radio. After another 20/25 mins or so, this started spreading to others, with the majority of players using the 152/117 either not being able to receive transmissions from anyone, or only getting inbound from select people regardless of line of sight. Some people tried to reconnect to the server to see if that would help, it did not and in some cases actually caused the loss of more systems, one individual even lost things like 343 and intercom with his pilot. At this point another solution that was attempted was reloading of loadouts to try and get a newly initialised radio which had no effect.

At this point I JIP'd into the mission about an hour after mission start, as a co-pilot. I was able to hear everyone, but only some people could receive from me, this is when i enabled both: `acre_sys_signal_showSignalHint = true;` and `ACRE_SIGNAL_DEBUGGING = 1;` to try and gather more information. My pilot was in a position of not being able to receive anything from anyone, on either his 343, 152 or 117. His signal hints were showing 0% whereas the same signal for me was reading at 90% or above on the same radio, in the same aircraft with him sitting next to me. During the next half an hour various pilots in particular picked up and dropped the ability to hear other people seemingly at random, one pilot in particular left the AO approx 4km away had disembarked and upon returning had lost everyone.

After about an hour and a half after mission start, most but not all callsigns including several aircraft gathered at a central location for a direct verbal brief, and after five our so minutes on the ground and remounting, people slowly started to gain the ability to hear each other, and for the last half an hour of the mission, most but not all regained the ability to receive transmissions again. 

For clarification this isn't like an extreme case of interference, due to terrain interception or anything most of these radios were on people in aircraft or simply sitting next to each other so LOS should be no issue, and it was not like things were coming through garbled it was just silence. The only way people new transmissions were being sent was due to the signal debug hint. 

I have access to server RPT and some client logs that were produced by the DLL after i turned on the signal debugging. If they are needed. It was a very strange circumstance, especially the fact that things started to come back after a while, we have no experienced that before, though more often than not when we run into it, vehicles are usually involved so that might be a place to investigate?

**Comment by jonpas** [comment]
@Twelvecaliber That sounds exactly what we had about 2 months ago on Anizay! Thanks for the very detailed description.

It would be great if you can upload as many logs of any kind as you can. This is pretty hard to debug even when it happens directly to us, which is too rare.

**Comment by Pironesis** [comment]
Server RPT: Mission in question starts line 17558

[arma3server_x64_2020-08-15_17-56-15.txt](https://github.com/IDI-Systems/acre2/files/5085896/arma3server_x64_2020-08-15_17-56-15.txt)


Client RPT from me as a JIP:

[arma3_x64_2020-08-15_21-15-19.txt](https://github.com/IDI-Systems/acre2/files/5085911/arma3_x64_2020-08-15_21-15-19.txt)

ACRE DLL Log file from me as a JIP:

[acre.log](https://github.com/IDI-Systems/acre2/files/5085926/acre.log)

Trimmed Client RPT from my pilot, was present for full duration of the mission, and at times was unable to receive any transmissions:

https://pastebin.com/cbfZ8nFd

We did have his ACRE DLL log file however it appears that variable that enables it stayed enabled on his client somewhere, and has since been wiped with client launches since that night. :( 

I might be able to get the DLL log file from others with issues if needed. @jonpas

**Comment by omar2205** [comment]
2.7.4.1032 fixed this problem for us

**Comment by MadSoloSniper** [comment]
That made me laugh . . . actually went to the workshop to see if ACRE2 had updated . . . . it has not. Guess ACRE is lost in "dev-no-mans-land" . . . maybe someday me hopes, given that it's a pretty major break, vs. just cosmetic or some obscure element.  Being able to comms is kinda the key to a comms system . . . just a thought.

PS - We wanna use ACRE, We like ACRE, we made a home out of ACRE, and now we're waiting on the magical "when it's done it's done" version to be released.  We've been able to recreate the "no comms / no carrier / just . . . . nothing" on multiple occasions, all within direct LOS of the individual on flat terrain different maps, some vanilla some community all not far enough from each other for actual "signal loss" (we're talking like 100M away).  We swap to 152s to make it work somewhat (still lose signal around 200M), then randomly at some obscure point it just fails again and random people will not be able to comms either inbound our outbound.  Makes no sense to any of us, we have RPT files that we've offered, and even have extended an offer for the DEVs to join us "in game" to experience it first hand . . . . but alas, all I see is random posts about how 1032 fixes all this and is yet to be released . . . . yawn.

**Comment by Koch** [comment]
We've tested 1032 with 50 people now, no issues that we've noticed. Some transmission issues between two distinct units, but they also had about 5km+ range between them with hilly terrain, so unlikely to be a mod issue per se.

**Comment by jonpas** [comment]
2.8.0 has been released. Please reopen if this (exact same) issue is encountered again, otherwise create a new issue for ease of tracking. Thank you all who tested!

**Comment by hotsumak4** [comment]
I have the problem that I can not transmit on the radio, but I repeat everything they talk ... I have already reinstalled both the mod and the TS3 and it continues as it can be? I report this happens when I am in the game alone I cannot transmit but I restore everything

**Comment by TheMagnetar** [comment]
This is a different issue. Try starting TS as an administrator

**Comment by coopermatt** [comment]
Our group is also experiencing this issue. We're on build  v2.9.0.1042. Radios will completely lose audio at 10 meter distances. The issue is random and happens with all the radio variants on/off the GSM spike. We've adjusted attenuation settings and still have no luck. 

Any updates on this one? I see it's closed but, still seems to be an issue.

**Comment by boulayb** [comment]
> Our group is also experiencing this issue. We're on build v2.9.0.1042. Radios will completely lose audio at 10 meter distances. The issue is random and happens with all the radio variants on/off the GSM spike. We've adjusted attenuation settings and still have no luck.
> 
> Any updates on this one? I see it's closed but, still seems to be an issue.

Same issue with our group. Version v2.9.1.
